### PR TITLE
Fix VanillaNetworkTest

### PR DESCRIPTION
**Pull request by amoudgl**
### Issue
CNN used to get stuck in local minima and couldn't converge in the given number of iterations. #922

### Hyperparameter Tuning
* Since the network is performing only binary classification task of recognizing 4 and 9, so number of output units in the CNN network has been updated to 2. This reduces the number of parameters in the network, hence avoiding wrong classification due to weights which could not be trained in the given number of iterations.
* Reduced number of units in full connected layers to reduce number of trainable parameters. 
* Reduced learning rate of optimizer to converge easily in given number of iterations. 

### Updates in PR
* Removed loop which performs the same test five times, hence reducing the testing time. 
* Renamed `error` variable as `correct` for more clarity because it counts the number of correct predictions by the network.
* Did Hyperparameter tuning as explained above.

### Results
Test was run for 75 times with `math::RandomSeed(std::time(NULL))` added to the top. Mean classification error over the runs was around `0.03` (rough estimate). Console log (with `classificationError` printed in each run) can be viewed [here](https://pastebin.com/1vBsHv3U). Please let me know if you are able to reproduce similar results.

### CODE REVIEW

**Comment by rcurtin** [comment]
Hi Abhinav,

Thanks for the detailed investigation.  It looks like maybe you and @zoq were working on this in parallel; I think that https://github.com/mlpack/mlpack/commit/ae93d749a2fed8c5418df14dbfbff9536177bdb6 solves the problem also.

I went ahead and tested both fixes for 1000 iterations to see if there were any failures; I saw no failures for either test strategy.  I also saw that there was no significant runtime difference for either strategy; it takes about 5.5s on my system for either strategy.

@zoq, @amoudgl: What do you think we should do here?  At the very least renaming `error` to `correct` is still a strategy that could work; maybe there are a couple other improvements, but in either case the big issue (the failing test) seems solved.

**Comment by amoudgl** [comment]
@rcurtin Yes, I saw that commit by @zoq. I made the changes with the thought in mind that why not do it in one shot like other tests rather than running a loop. I also hoped that hyperparameter tuning would reduce the testing time but thanks for your analysis, I didn't actually check the testing time difference. I see that it doesn't make any difference so we can go with either solution. I will try to perform multiple experiments with multiple iterations in each one and come up with some detailed analysis of time and number of failures, to converge to one solution.

**Comment by zoq** [comment]
If @amoudgl solution runs for 1000 iterations without any error I think we should go with it instead of running the test again. As I tested my solution I noticed that in like 5 of 100 cases the tests runs a second time. @amoudgl what do you think?

**Comment by amoudgl** [comment]
@zoq Thanks for the info. Yes, I agree with your view point. Please let me know if I should make any edits in my solution otherwise I think it is merge ready.

File: src/mlpack/tests/convolutional_network_test.cpp [hunk 1]
```diff
@@ -47,11 +47,13 @@ BOOST_AUTO_TEST_CASE(VanillaNetworkTest)
   {
     if (i < nPoints / 2)
     {
-      Y(i) = 4;
+      // class 1 - digit = 4
```
Comment (zoq) [inline src/mlpack/tests/convolutional_network_test.cpp]: Pedantic style issue  it's helpful to always have comments that are complete sentences with proper grammar and punctuation.

Comment (amoudgl) [inline src/mlpack/tests/convolutional_network_test.cpp]: Oops, thanks for pointing this out. Fixed now. 👍

**Comment by amoudgl** [comment]
@zoq I updated the PR to reflect code review changes. Do you think this PR is ready to be merged now?
